### PR TITLE
GL Journal End of Year

### PR DESCRIPTION
**Pull request by metas-rc**
https://github.com/metasfresh/metasfresh/issues/10964

### CODE REVIEW

File: backend/de.metas.acct.base/src/main/java/de/metas/acct/gljournal/IGLJournalDAO.java [hunk 1]
```diff
@@ -52,4 +53,6 @@
 	 * @return
 	 */
 	List<I_GL_Journal> retrievePostedWithoutFactAcct(Properties ctx, Date startTime);
+
+	DocTypeId retrieveDocTypeGLJournal(int clientId, int orgId);
```
Comment (teosarca) [inline backend/de.metas.acct.base/src/main/java/de/metas/acct/gljournal/IGLJournalDAO.java:57]: move it to IGLJournalBL.
use ClientId, OrgId

File: backend/de.metas.acct.base/src/main/java/de/metas/acct/gljournal/GL_Journal_Builder.java [hunk 2]
```diff
@@ -43,50 +41,41 @@
...
 	private final transient ICurrencyDAO currencyDAO = Services.get(ICurrencyDAO.class);
+	private final IGLJournalDAO glJournalDAO = Services.get(IGLJournalDAO.class);
 
-	private final I_GL_JournalBatch glJournalBatch;
 	private final I_GL_Journal glJournal;
 
 	private final List<GL_JournalLine_Builder> glJournalLineBuilders = new ArrayList<>();
 
-	private GL_Journal_Builder(final I_GL_JournalBatch glJournalBatch)
+	private GL_Journal_Builder(final GLJournalRequest request)
```
Comment (teosarca) [inline backend/de.metas.acct.base/src/main/java/de/metas/acct/gljournal/GL_Journal_Builder.java:59]: makes no sense having this "partial" request here. 
Pls have all the required values in the request 
or introduce a setter method for each of them.

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.acct.base/src/main/java/de/metas/acct/gljournal/GL_Journal_Builder.java (hunk 2 above)
Comment (metas-rc) [reply backend/de.metas.acct.base/src/main/java/de/metas/acct/gljournal/GL_Journal_Builder.java:57]: Done. I chose to get rid of the setters and use the request.

**Comment by metas-rc** [comment]
Additionally, I am no longer using the currency asa parameter anymore because it was always taken form the accounting schema. I'm going to do the same

**Pull request merged**